### ISSUE TITLE
`querySelectorAll` fails for elements with IDs that contain special characters

### ISSUE THREAD

**Issue by zzzzBov** [oddbird/css-anchor-positioning#156]
The value of the `anchor` attribute is used as-is without properly being escaped for a query selector, which causes errors with IDs that use special characters (such as are generated by React's `useId()` hook):

```
validate.ts:142 Uncaught (in promise) DOMException: Failed to execute 'querySelectorAll' on 'Document': '#:R4p:' is not a valid selector.
```

Updating

https://github.com/oddbird/css-anchor-positioning/blob/14e560a7ea375de71ad717c22c97b4103fe7ea85/src/parse.ts#L452

to something like

```
`#${anchorAttr.replaceAll(/([^a-z0-9_-])/gi, '\\$1')}`
```

should fix the issue, although you may want to double check the specific characters that need to be escaped for `querySelector`.

**Comment by jgerigmeyer** [comment]
@zzzzBov Thanks! I'll try to get a fix in this week, though I'd also be happy to review a PR if you want to submit one.

**Comment by jgerigmeyer** [comment]
Released in [v0.0.5](https://github.com/oddbird/css-anchor-positioning/releases/tag/v0.0.5)

**Comment by zzzzBov** [comment]
@jgerigmeyer thanks for the fix, and also TIL that `CSS.escape` exists 👍
